src/components/Uploader/index.js

```diff
@@ -2,29 +2,39 @@ import { h } from 'preact'
 import Dropzone from 'react-dropzone'
 import theme from '../Theme/style.css'
 import style from './style.css'
+import { isDesktop } from '../utils'
 import {errors} from '../strings/errors'
 import { trackComponentAndMode } from '../../Tracker'
 import SwitchDevice from '../crossDevice/SwitchDevice'
 import { mobileCopy, desktopCopy } from '../strings/uploadCopy'
 
-const instructionsCopy = (method, side, allowCrossDeviceFlow) => {
-  const instructions = allowCrossDeviceFlow ? desktopCopy.instructions : mobileCopy.instructions
+const instructionsCopy = (method, side) => {
+  const instructions = isDesktop ? desktopCopy.instructions : mobileCopy.instructions
   return method === 'document' ? instructions[method][side] : instructions[method]
 }
 
-const UploadInstructions = ({error, method, side, allowCrossDeviceFlow}) =>
-  <div className={style.base}>
-    <span className={`${theme.icon} ${style.icon}`}></span>
-    <p className={style.text}>{instructionsCopy(method, side, allowCrossDeviceFlow)}</p>
-    <UploadError error={errors[error.name]} />
-  </div>
+const instructionsIcon = () =>
+  isDesktop ? style.uploadIcon : style.cameraIcon
+
+const UploadInstructions = ({error, method, side}) => {
+  const noError = Object.keys(error).length === 0 && error.constructor === Object
+  return (
+    <div className={style.base}>
+      <span className={`${theme.icon} ${instructionsIcon()}`}></span>
+      { noError ? <Instructions side={side} method={method} /> :
```

Comment:
Why do you need this line? This is breaking when you try to open the cross device client.

Reply:
This is so we display either instructions or the error (per Daniel's suggestion). It was breaking cross device as it was being overwritten by the error property from the crossDeviceRouter. I have renamed that property so the bug is fixed.